### PR TITLE
Hal API to retrieve TID stats via iw and Netlink, Adding Hal test binary

### PR DESCRIPTION
**Pull request by Rajakumaran1106**
Signed-off-by: Rajakumaran1106 <rajakumaran.a@ltts.com>

### CODE REVIEW

File: source/wifi/Makefile.am [hunk 1]
```diff
@@ -28,3 +28,8 @@ lib_LTLIBRARIES = libhal_wifi.la
 libhal_wifi_la_CPPFLAGS = -I$(top_srcdir)/./include $(CPPFLAGS)
```
Comment (manigandanj) [inline source/wifi/Makefile.am]: remove this "./"

File: source/wifi/Makefile.am [hunk 2]
```diff
@@ -28,3 +28,8 @@ lib_LTLIBRARIES = libhal_wifi.la
...
 libhal_wifi_la_SOURCES = wifi_hal.c
 libhal_wifi_la_LDFLAGS =
+
+#wifihal test binary
+bin_PROGRAMS = wifihal
+wifihal_CPPFLAGS = -I$(top_srcdir)/include $(CPPFLAGS)
```
Comment (manigandanj) [inline source/wifi/Makefile.am]: Add -DWIFI_HAL_TEST in CPPFLAGS

File: source/wifi/wifi_hal.c [hunk 3]
```diff
@@ -18,6 +18,7 @@
 */
 
 #define HAL_NETLINK_IMPL
+#define WIFI_HAL_TEST
```
Comment (manigandanj) [inline source/wifi/wifi_hal.c]: Must remove this line.

File: source/wifi/wifi_hal.c [hunk 4]
```diff
@@ -6680,10 +6701,178 @@ INT wifi_setRadioStatsEnable(INT radioIndex, BOOL enable)
...
+}
+#endif
+
 INT wifi_getApAssociatedDeviceTidStatsResult(INT radioIndex,  mac_address_t *clientMacAddress, wifi_associated_dev_tid_stats_t *tid_stats,  ULLONG *handle)
 {
-    // TODO Implement me!
-    return RETURN_ERR;
+#ifdef HAL_NETLINK_IMPL
+    Netlink nl;
+    char  if_name[10];
```
Comment (manigandanj) [inline source/wifi/wifi_hal.c:6766]: one space is enough

**Review by manigandanj — CHANGES_REQUESTED** [review]

**Comment by manigandanj** [comment]
Looks okay to me. It can be merged.

**Review by rajkumar154 — APPROVED** [review]

**Pull request merged**